code/game/objects/items/weapons/storage/bags.dm

```diff
@@ -320,20 +321,22 @@
 
 	// Make each item scatter a bit
 	for(var/obj/item/I in oldContents)
-		spawn()
-			for(var/i = 1, i <= rand(1,2), i++)
-				if(I)
-					step(I, pick(NORTH,SOUTH,EAST,WEST))
-					sleep(rand(2,4))
+		I.forceMove(M)
+		INVOKE_ASYNC(src, PROC_REF(scatter_tray_items), I)
 
 	if(prob(50))
 		playsound(M, 'sound/items/trayhit1.ogg', 50, 1)
 	else
 		playsound(M, 'sound/items/trayhit2.ogg', 50, 1)
 
-	if(ishuman(M))
-		if(prob(10))
-			M.Weaken(4 SECONDS)
+	if(ishuman(M) && prob(10))
+		M.KnockDown(4 SECONDS)
+
+/obj/item/storage/bag/tray/proc/scatter_tray_items(obj/item/I)
+	if(I)
```

Comment:
Early return this, and fix the spacing between directions and numbers, please

Reply:
Done.